Documentation/postmortems/v3.5-data-inconsistency.md

```diff
@@ -0,0 +1,138 @@
+# v3.5 data inconsistency postmortem
+
+|         |            |
+|---------|------------|
+| Authors | serathius@ |
+| Date    | 2022-04-20 |
+| Status  | draft      |
+
+## Summary
+
+|         |                                                                                                                                                                                                                               |
+|---------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Summary | Code refactor resulted in consistent index not being saved atomically, allowing for crash causing inconsistency with other cluster members.                                                                                   |
+| Impact  | No user reported problems in production as triggering the issue required frequent crashes, however issue was critical enough to motivate a public statement. Main impact comes from loosing user trust into etcd reliability. |
+
+## Background
+
+Etcd v3 state is preserved on disk in two forms write ahead log (WAL) and database state (DB).
```

Comment:
etcd instead of Etcd :P. We use etcd even if it is the first word in the sentence conventionally.

Reply:
:) +1 @xiang90 Good catch!

Reply:
I now noticed it at few other places :) so should do a clean sweep @serathius Thanks!